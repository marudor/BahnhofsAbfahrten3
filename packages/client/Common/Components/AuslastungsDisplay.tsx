import * as React from 'react';
import SingleAuslastungsDisplay from 'client/Common/Components/SingleAuslastungsDisplay';
import styled from 'styled-components/macro';
import type { Route$Auslastung } from 'types/routing';

const Wrap = styled.div`
  display: flex;
  margin-bottom: 0.3em;
`;

const Entry = styled.div`
  display: flex;
  flex-direction: column;
  margin-right: 0.5em;
  align-items: center;
`;

export interface Props {
  auslastung: Route$Auslastung;
}

const AuslastungsDisplay = (props: Props) => {
  const { auslastung } = props;

  return (
    <Wrap data-testid="auslastungDisplay">
      <Entry data-testid="first">
        <span>1</span>
        <SingleAuslastungsDisplay auslastung={auslastung.first} />
      </Entry>
      <Entry data-testid="second">
        <span>2</span>
        <SingleAuslastungsDisplay auslastung={auslastung.second} />
      </Entry>
    </Wrap>
  );
};

export default AuslastungsDisplay;
