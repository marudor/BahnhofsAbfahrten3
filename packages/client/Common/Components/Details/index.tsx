import { DetailsContext } from './DetailsContext';
import { format } from 'date-fns';
import { getDetails } from 'client/Common/service/details';
import { Header } from './Header';
import { StopList } from './StopList';
import { useEffect, useState } from 'react';
import { useHeaderTagsActions } from 'client/Common/provider/HeaderTagProvider';
import { useQuery } from 'client/Common/hooks/useQuery';
import type { AxiosError } from 'axios';
import type { ParsedSearchOnTripResponse } from 'types/HAFAS/SearchOnTrip';

interface Props {
  train: string;
  stationId?: string;
  initialDeparture?: string;
  currentStopId?: string;
  urlPrefix?: string;
}

export const Details = ({
  train,
  initialDeparture,
  stationId,
  currentStopId,
  urlPrefix,
}: Props) => {
  const query = useQuery();
  const [details, setDetails] = useState<ParsedSearchOnTripResponse>();
  const [error, setError] = useState<AxiosError>();
  const { updateTitle, updateDescription } = useHeaderTagsActions();

  useEffect(() => {
    if (details) {
      updateTitle(
        `${details.train.name} @ ${format(
          details.departure.time,
          'dd.MM.yyyy'
        )}`
      );
    } else {
      updateTitle(train);
    }
    let description = `Details zu ${train}`;

    if (initialDeparture) {
      description += ` am ${format(
        Number.parseInt(initialDeparture, 10),
        'dd.MM.yyyy'
      )}`;
    }
    updateDescription(description);
  }, [details, initialDeparture, train, updateDescription, updateTitle]);

  useEffect(() => {
    setDetails(undefined);
    getDetails(
      train,
      initialDeparture,
      currentStopId,
      stationId,
      query.profile as any
    )
      .then((details) => {
        setDetails(details);
        setError(undefined);
      })
      .catch((e) => {
        setError(e);
      });
  }, [train, initialDeparture, currentStopId, query.profile, stationId]);

  return (
    <DetailsContext.Provider
      value={{
        details,
        error,
        urlPrefix,
      }}
    >
      <Header train={train} />
      <StopList />
    </DetailsContext.Provider>
  );
};
