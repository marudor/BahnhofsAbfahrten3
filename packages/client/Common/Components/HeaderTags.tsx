import { Helmet } from 'react-helmet-async';
import { useLocation } from 'react-router';
import HeaderTagContainer from 'client/Common/container/HeaderTagContainer';

const HeaderTags = () => {
  const { title, description } = HeaderTagContainer.useContainer();
  const url = `${global.BASE_URL}${useLocation().pathname}`;
  const image = `${global.BASE_URL}/android-chrome-384x384.png`;

  return (
    <Helmet>
      <title>{title}</title>
      <link data-testid="canonicalLink" rel="canonical" href={url} />
      <meta name="description" content={description} />
      {/* Twitter Start */}
      <meta name="twitter:card" content="summary" />
      <meta name="twitter:site" content="@marudor" />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:creator" content="@marudor" />
      <meta name="twitter:image" content={image} />
      {/* Twitter End */}
      {/* Open Graph Start */}
      <meta property="og:title" content={title} />
      <meta property="og:type" content="website" />
      <meta property="og:description" content={description} />
      <meta property="og:image" content={image} />
      <meta property="og:url" content={url} />
      <meta property="og:locale" content="de_DE" />
      {/* Open Graph End */}
    </Helmet>
  );
};

export default HeaderTags;
