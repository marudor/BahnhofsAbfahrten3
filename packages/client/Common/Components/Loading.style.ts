import { makeStyles } from '@material-ui/styles';

export default makeStyles((theme) => ({
  absolute: {
    position: 'absolute',
  },
  '@keyframes cube': {
    '0%,70%,100%': {
      transform: 'scale3D(1, 1, 1)',
    },
    '35%': {
      transform: 'scale3D(0, 0, 1)',
    },
  },

  cube: {
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    width: '35vmin',
    height: '35vmin',
    margin: 'auto',
    '& > div': {
      width: '33%',
      height: '33%',
      backgroundColor: theme.palette.text.primary,
      float: 'left',
      animation: '$cube 1.3s infinite ease-in-out',
    },
    '& > div:nth-child(1)': { animationDelay: '0.2s' },
    '& > div:nth-child(2)': { animationDelay: '0.3s' },
    '& > div:nth-child(3)': { animationDelay: '0.4s' },
    '& > div:nth-child(4)': { animationDelay: '0.1s' },
    '& > div:nth-child(5)': { animationDelay: '0.2s' },
    '& > div:nth-child(6)': { animationDelay: '0.3s' },
    '& > div:nth-child(7)': { animationDelay: '0s' },
    '& > div:nth-child(8)': { animationDelay: '0.1s' },
    '& > div:nth-child(9)': { animationDelay: '0.2s' },
  },

  '@keyframes dots': {
    '0%': {
      top: 6,
      height: 51,
    },
    '50%,100%': {
      top: 19,
      height: 26,
    },
  },

  dots: {
    display: 'inline-block',
    position: 'relative',
    width: 64,
    height: 64,
    '& > div': {
      display: 'inline-block',
      position: 'absolute',
      left: 6,
      width: 13,
      background: theme.palette.text.primary,
      animation: '$dots 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite',
    },
    '& > div:nth-child(1)': {
      left: 6,
      animationDelay: '-0.24s',
    },
    '& > div:nth-child(2)': {
      left: 26,
      animationDelay: '-0.12s',
    },
    '& > div:nth-child(3)': {
      left: 45,
      animationDelay: '0',
    },
  },
}));
