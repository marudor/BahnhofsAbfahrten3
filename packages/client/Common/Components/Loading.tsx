import { makeStyles } from '@material-ui/core';
import clsx from 'clsx';
import type { FC, ReactElement } from 'react';

const useStyles = makeStyles((theme) => ({
  absolute: {
    position: 'absolute',
  },
  '@keyframes grid': {
    '0%,70%,100%': {
      transform: 'scale3D(1, 1, 1)',
    },
    '35%': {
      transform: 'scale3D(0, 0, 1)',
    },
  },
  grid: {
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    width: '35vmin',
    height: '35vmin',
    margin: 'auto',
    '& > div': {
      width: '33%',
      height: '33%',
      backgroundColor: theme.palette.text.primary,
      float: 'left',
      animation: '$grid 1.3s infinite ease-in-out',
    },
    '& > div:nth-child(1)': { animationDelay: '0.2s' },
    '& > div:nth-child(2)': { animationDelay: '0.3s' },
    '& > div:nth-child(3)': { animationDelay: '0.4s' },
    '& > div:nth-child(4)': { animationDelay: '0.1s' },
    '& > div:nth-child(5)': { animationDelay: '0.2s' },
    '& > div:nth-child(6)': { animationDelay: '0.3s' },
    '& > div:nth-child(7)': { animationDelay: '0s' },
    '& > div:nth-child(8)': { animationDelay: '0.1s' },
    '& > div:nth-child(9)': { animationDelay: '0.2s' },
  },
  '@keyframes dots': {
    '0%': {
      top: 6,
      height: 51,
    },
    '50%,100%': {
      top: 19,
      height: 26,
    },
  },
  dots: {
    display: 'inline-block',
    position: 'relative',
    width: 64,
    height: 64,
    '& > div': {
      display: 'inline-block',
      position: 'absolute',
      left: 6,
      width: 13,
      background: theme.palette.text.primary,
      animation: '$dots 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite',
    },
    '& > div:nth-child(1)': {
      left: 6,
      animationDelay: '-0.24s',
    },
    '& > div:nth-child(2)': {
      left: 26,
      animationDelay: '-0.12s',
    },
    '& > div:nth-child(3)': {
      left: 45,
      animationDelay: '0',
    },
  },
}));

interface Props {
  isLoading?: boolean;
  className?: string;
  type?: LoadingType;
  relative?: boolean;
  children?: ReactElement;
}

export const enum LoadingType {
  grid,
  dots,
}

const InnerLoading = ({ type, relative }: Pick<Props, 'type' | 'relative'>) => {
  const classes = useStyles();
  switch (type) {
    default:
    case LoadingType.grid:
      return (
        <div
          className={clsx(classes.grid, !relative && classes.absolute)}
          data-testid="grid"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      );
    case LoadingType.dots:
      return (
        <div className={classes.dots} data-testid="dots">
          <div />
          <div />
          <div />
        </div>
      );
  }
};

export const Loading: FC<Props> = ({
  isLoading,
  className,
  children,
  relative = false,
  type = LoadingType.grid,
}) => {
  if (isLoading || !children) {
    return (
      <div data-testid="loading" className={className}>
        <InnerLoading type={type} relative={relative} />
      </div>
    );
  }

  return children;
};
