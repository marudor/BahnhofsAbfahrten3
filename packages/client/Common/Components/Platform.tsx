import { cancelledCss, changedCss } from 'client/util/cssUtils';
import styled from 'styled-components';

interface Props {
  className?: string;
  cancelled?: boolean;
  scheduled?: string;
  real?: string;
}

const Wrap = styled.div<{ cancelled?: boolean; changed?: boolean }>`
  ${({ cancelled, changed }) => [
    cancelled && cancelledCss,
    changed && changedCss,
  ]}
`;

const ChangedWrapper = styled.span`
  padding-left: 0.3em;
  ${cancelledCss}
`;

export const Platform = ({ className, cancelled, scheduled, real }: Props) => {
  const changed = Boolean(scheduled && scheduled !== real);

  return (
    <Wrap changed={changed} cancelled={cancelled} className={className}>
      <span data-testid="real">{real}</span>
      {changed && (
        <ChangedWrapper data-testid="scheduled">({scheduled})</ChangedWrapper>
      )}
    </Wrap>
  );
};
