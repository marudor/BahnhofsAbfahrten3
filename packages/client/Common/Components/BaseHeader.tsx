import { AppBar, IconButton, Toolbar } from '@material-ui/core';
import { ReactNode, useContext } from 'react';
import ActionMenu from '@material-ui/icons/Menu';
import HeaderTags from './HeaderTags';
import NavigationContext from 'client/Common/Components/Navigation/NavigationContext';

interface Props {
  children: ReactNode;
}

const BaseHeader = ({ children }: Props) => {
  const { toggleDrawer } = useContext(NavigationContext);

  return (
    <AppBar position="fixed" data-testid="header">
      <HeaderTags />
      <Toolbar disableGutters>
        <IconButton
          data-testid="navToggle"
          aria-label="Menu"
          onClick={toggleDrawer}
          color="inherit"
        >
          <ActionMenu color="inherit" />
        </IconButton>
        {children}
      </Toolbar>
    </AppBar>
  );
};

export default BaseHeader;
