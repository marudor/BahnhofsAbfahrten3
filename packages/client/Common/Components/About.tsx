import { BaseHeader } from 'client/Common/Components/BaseHeader';
import { BugReport, Extension, Message } from '@material-ui/icons';
import { Button, makeStyles } from '@material-ui/core';
import type { FC } from 'react';

const Privacy = () => (
  <div data-testid="Privacy">
    <h2>Impressum</h2>
    <p>Verantwortlich für diese Seite ist:</p>
    <p>
      <span>
        {globalThis.IMPRINT.name}
        <br />
        {globalThis.IMPRINT.street}
        <br />
        {globalThis.IMPRINT.town}
      </span>
    </p>
    <h2>Datenschutzerklärung</h2>
    <h3>Logdaten des Webservers</h3>
    <p>
      Bei jedem Aufruf dieser Website werden vom Webserver die folgenden Daten
      gespeichert und für zwei Wochen aufbewahrt:
    </p>
    <p>
      Datum und Uhrzeit,
      <br /> die aufgerufene Seite,
      <br /> der dabei verwendete Browser (gemäß des von diesem gesendeten
      UserAgents) und
      <br /> der Referer (d.h. die auf diese Seite verlinkende Website —
      ebenfalls gemäß der vom Browser gesendeten Informationen, kann meist im
      Browser deaktiviert werden).
      <br />
      Die IP-Adresse sowie die bei Aufruf der Hauptseite an den Server
      übertragenen Geokoordinaten werden nicht gespeichert.
    </p>
  </div>
);

const useStyles = makeStyles((theme) => ({
  wrap: {
    margin: `${theme.shape.headerSpacing}px 10px 0`,
    display: 'flex',
    flexDirection: 'column',
  },
  donation: {
    marginTop: 5,
  },
  buttonsWrap: {
    [theme.breakpoints.down('md')]: {
      height: '10em',
    },
  },
  buttons: {
    '& svg': {
      marginRight: 5,
    },
    display: 'flex',
    justifyContent: 'space-around',
    marginTop: '1em',
    [theme.breakpoints.down('md')]: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
    },
  },
}));

export const About: FC = () => {
  const classes = useStyles();
  return (
    <>
      <BaseHeader>About</BaseHeader>
      <div className={classes.wrap}>
        <span>
          Entwickelt von{' '}
          <a
            href="https://twitter.com/marudor"
            target="_blank"
            rel="noopener noreferrer"
          >
            @marudor
          </a>{' '}
          /{' '}
          <a
            href="https://chaos.social/@marudor"
            target="_blank"
            rel="noopener noreferrer"
          >
            @marudor@chaos.social
          </a>
        </span>
        <div className={classes.donation}>
          Falls euch der Service gefällt, könnt ihr mir mit folgendem Button per
          PayPal Geld spenden. Falls ihr anderweitig spenden wollt, schreibt mir
          &apos;ne Mail an
          <a href="mailto:spende@marudor.de"> spende@marudor.de</a>
        </div>
        <div className={classes.buttonsWrap}>
          <div className={classes.buttons}>
            <a
              href="https://paypal.me/marudor"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button variant="outlined">Paypal Spende</Button>
            </a>
            <a
              href="https://twitter.com/marudor"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button variant="outlined">
                <Message />
                Kontakt
              </Button>
            </a>
            <a
              href="https://github.com/marudor/BahnhofsAbfahrten/issues/new?assignees=&labels=bug&template=bug_report.md&title="
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button variant="outlined">
                <BugReport />
                Bugs?
              </Button>
            </a>
            <a
              href="https://github.com/marudor/BahnhofsAbfahrten/issues/new?assignees=&labels=feature&template=feature_request.md&title="
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button variant="outlined">
                <Extension />
                Features?
              </Button>
            </a>
          </div>
        </div>
        <Privacy />
      </div>
    </>
  );
};
// eslint-disable-next-line import/no-default-export
export default About;
