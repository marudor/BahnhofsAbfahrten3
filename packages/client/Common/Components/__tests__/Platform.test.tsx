import { Platform } from 'client/Common/Components/Platform';
import { render } from 'client/__tests__/testHelper';

describe('Platform', () => {
  it('No platform provided', () => {
    const { container, queryByTestId } = render(Platform);

    expect(queryByTestId('real')).toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
  it('scheduled === real', () => {
    const { container, queryByTestId } = render(Platform, {
      real: '1',
      scheduled: '1',
    });

    expect(queryByTestId('real')).toHaveTextContent('1');
    expect(queryByTestId('scheduled')).toBeNull();
    expect(container).toMatchSnapshot();
  });
  it('scheduled === real & cancelled', () => {
    const { container, queryByTestId } = render(Platform, {
      real: '1',
      scheduled: '1',
      cancelled: true,
    });

    expect(queryByTestId('real')).toHaveTextContent('1');
    expect(container).toHaveStyle('text-decoration: line-through');
    expect(container).toMatchSnapshot();
  });
  it('scheduled !== real & cancelled', () => {
    const { container, theme, queryByTestId } = render(Platform, {
      real: '1',
      scheduled: '2',
      cancelled: true,
    });

    expect(queryByTestId('real')).toHaveTextContent('1');
    expect(container).toHaveStyle(`
    text-decoration: line-through;
    color: ${theme.colors.red};`);
    expect(container).toMatchSnapshot();
  });
  it('scheduled !== real', () => {
    const { container, theme, queryByTestId } = render(Platform, {
      real: '1',
      scheduled: '2',
    });

    expect(queryByTestId('real')).toHaveTextContent('1');
    expect(queryByTestId('scheduled')).toHaveTextContent('2');
    expect(queryByTestId('scheduled')).toHaveStyle(
      'text-decoration: line-through;',
    );
    expect(container).not.toHaveStyle(`text-decoration: line-through;`);
    expect(container).toHaveStyle(`color: ${theme.colors.red};`);
    expect(container).toMatchSnapshot();
  });
});
