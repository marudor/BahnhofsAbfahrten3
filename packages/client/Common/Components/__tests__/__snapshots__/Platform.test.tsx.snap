// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform No platform provided 1`] = `
<div
  class=""
  data-testid="platform"
>
  <span
    data-testid="real"
  />
</div>
`;

exports[`Platform scheduled !== real & cancelled 1`] = `
<div
  class="Platform-cancelled Platform-changed"
  data-testid="platform"
>
  <span
    data-testid="real"
  >
    1
  </span>
  <span
    class="Platform-changedWrapper"
    data-testid="scheduled"
  >
    (
    2
    )
  </span>
</div>
`;

exports[`Platform scheduled !== real 1`] = `
<div
  class="Platform-changed"
  data-testid="platform"
>
  <span
    data-testid="real"
  >
    1
  </span>
  <span
    class="Platform-changedWrapper"
    data-testid="scheduled"
  >
    (
    2
    )
  </span>
</div>
`;

exports[`Platform scheduled === real & cancelled 1`] = `
<div
  class="Platform-cancelled"
  data-testid="platform"
>
  <span
    data-testid="real"
  >
    1
  </span>
</div>
`;

exports[`Platform scheduled === real 1`] = `
<div
  class=""
  data-testid="platform"
>
  <span
    data-testid="real"
  >
    1
  </span>
</div>
`;
